src/components/Country.js

```diff
@@ -0,0 +1,36 @@
+import React from 'react';
+import { useNavigate } from 'react-router-dom';
+
+const Country = ({ data }) => {
+  const navigate = useNavigate();
+
+  const handleRoute = () => {
+    navigate(`/details/${data.name}`, { state: data });
+  };
+
+  return (
+    <button type="button" onClick={handleRoute} className="country">
+      <img className="country-img" src={data.flag} alt="Country Flag" />
+      <div className="details">
+        <h2>{data.name}</h2>
+        <p>
+          Population:
+          <span>{data.population}</span>
+        </p>
+        <p>
+          Continent:
+          <span>{data.region}</span>
+        </p>
+        <p>
+          Capital:
+          <span>{data.capital}</span>
+        </p>
+        {/* <p>.</p>
+      <p>.</p>
+      <p>click for more details</p> */}
+      </div>
```

Comment:
- [ ] Please do not leave unused code as a comment in your codebase. this will make your code much noisier, It is always a good practice to keep your code clean from such comments. 😉 👍🏻

Reply:
- [ ] [OPTIONAL] Kindly remove the unused lines of code as recommended by the previous reviewer for a clean code base.